### PR TITLE
added validation for sl price higher than next price

### PR DESCRIPTION
**Pull request by johnnieskywalker**
# [Title](https://shortcutLink)

Redirecting to close vault if next price is lower than selected stop-loss
  
## Changes 👷‍♀️
  <Please add short list of changes>
- item 1
  
## How to test 🧪
  <Please explain how to test your changes>
- step 1 ...

### CODE REVIEW

File: features/automation/protection/common/validation.ts [hunk 1]
```diff
@@ -46,14 +49,18 @@ export function errorsValidation({
...
 }: {
   selectedSLValue: BigNumber
   ilkData: IlkData
   txError?: TxError
+  collateralizationRatioAtNextPrice: BigNumber
 }) {
   const insufficientEthFundsForTx = ethFundsForTxValidator({ txError })
 
-  const stopLossOnNearLiquidationRatio = slCollRatioNearLiquidationRatio(selectedSLValue, ilkData)
+  const stopLossOnNearLiquidationRatio =
```
Comment (piekczyk) [inline features/automation/protection/common/validation.ts:61]: I think this condition will still return true for case when user tries to setup SL near to liquidation ratio even though next price will be much higher, in other words it won't fix the issue that Sam noticed

**Comment by johnnieskywalker** [comment]
Hardcoded collateralizationRatioAtNextPrice as 500, when I'm trying to set sl at 504, I get the following message

image

File: features/automation/protection/common/validation.ts (hunk 1 above)
Comment (johnnieskywalker) [reply features/automation/protection/common/validation.ts]: commented it out with todo remark to bring it back later

File: features/automation/protection/controls/AdjustSlFormControl.tsx [hunk 2]
```diff
@@ -239,11 +244,17 @@ export function AdjustSlFormControl({
...
-    translationKey: slCollRatioNearLiquidationRatio(selectedSLValue, ilkData)
-      ? 'close-vault'
-      : isStopLossEnabled
-      ? 'update-stop-loss'
-      : 'add-stop-loss',
+    translationKey:
+      slCollRatioNearLiquidationRatio(selectedSLValue, ilkData) ||
+      slRatioHigherThanCurrentOrNext(
+        selectedSLValue,
+        collateralizationRatioAtNextPrice,
```
[the remark this answers is not in the document]
Comment (johnnieskywalker) [reply features/automation/protection/controls/AdjustSlFormControl.tsx:251]: AdjustSlFormLayout line 402*

**Comment by johnnieskywalker** [comment]
In order to test next price coll ratio must be mocked in 2 places
`AdjustSlFormLayout` line 228 for ex in this case as 
`(slCollRatioNearLiquidationRatio(selectedSLValue, ilkData) ||
            slRatioHigherThanCurrentOrNext(
              selectedSLValue,
              new BigNumber(5),
              currentCollateralRatio,
            ))
            `
            
            ProtectionFormControl line 105 
            `                  collateralizationRatioAtNextPrice={new BigNumber(5)}
`

expected outcome :

image

**Review by piekczyk — DISMISSED** [review]

**Review by piekczyk — APPROVED** [review]

**Pull request merged**